hybrid/decomposers.py

```diff
@@ -44,6 +45,91 @@ class IdentityDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
     def next(self, state, **runopts):
         return state.updated(subproblem=state.problem)
 
+class ComponentDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
+    """Selects a subproblem of variables that make up a connected component.
+
+    Args:
+        rolling (bool, optional, default=True):
+            If True, successive calls for the same problem (with possibly
+            different samples) produce subproblems on different components,
+            selected by rolling down the list of all components sorted by
+            the `key` function.
+
+        key (function, optional, default=len):
+            Function used to extract a comparison key from each component of 
+            the problem. The comparison keys determine which component will be 
+            used to build the subproblem. By default, the components are sorted 
+            by size in decreasing order, so the largest component will be used first.
+        
+        reverse (bool, optional, default=True):
```

Comment:
I'd suggest we default to `key=None` and `reverse=None`, meaning no sorting by default. See my comments below on iteration.